tests/MongoDB.Driver.Core.Tests/Specifications/server-discovery-and-monitoring/TestRunner.cs

```diff
@@ -282,38 +271,26 @@ private ICluster BuildCluster(BsonDocument definition)
                 .CreateCluster();
         }
 
-        private class TestCaseFactory : IEnumerable<object[]>
+        private class TestCaseFactory : JsonDrivenTestCaseFactory
         {
-            public IEnumerator<object[]> GetEnumerator()
+            #region static
+            private static readonly string[] __ignoredTestNames =
             {
-                const string prefix = "MongoDB.Driver.Core.Tests.Specifications.server_discovery_and_monitoring.tests.";
-                const string monitoringPrefix = "MongoDB.Driver.Core.Tests.Specifications.server_discovery_and_monitoring.tests.monitoring.";
-                var executingAssembly = typeof(TestCaseFactory).GetTypeInfo().Assembly;
-                var enumerable = executingAssembly
-                    .GetManifestResourceNames()
-                    .Where(path => path.StartsWith(prefix) && path.EndsWith(".json"))
-                    .Where(path => !path.StartsWith(monitoringPrefix))
-                    .Select(path => ReadDefinition(path))
-                    .Select(definition => new object[] { definition });
-                return enumerable.GetEnumerator();
-            }
+                "MongoDB.Driver.Core.Tests.Specifications.server_discovery_and_monitoring.tests.monitoring."
+            };
+            #endregion
 
-            IEnumerator IEnumerable.GetEnumerator()
-            {
-                return GetEnumerator();
-            }
+            protected override string PathPrefix => "MongoDB.Driver.Core.Tests.Specifications.server_discovery_and_monitoring.tests.";
 
-            private static BsonDocument ReadDefinition(string path)
+            protected override IEnumerable<JsonDrivenTestCase> CreateTestCases(BsonDocument document)
             {
-                var executingAssembly = typeof(TestCaseFactory).GetTypeInfo().Assembly;
-                using (var definitionStream = executingAssembly.GetManifestResourceStream(path))
-                using (var definitionStringReader = new StreamReader(definitionStream))
+                var path = document["_path"].ToString();
+                if (__ignoredTestNames.Any(i => path.Contains(i)))
```

Comment:
I think this would be a bit more readable (and precise) using a more descriptive variable than `i` and `StartsWith` instead of `Contains`.

    if (!__ignoredTestFolders.Any(ignoredFolder => path.StartsWith(ignoredFolder)))

Reply:
done